File: src/main/java/com/polygloat/dtos/request/validators/exceptions/ValidationException.java
package com.polygloat.dtos.request.validators.exceptions;

import com.polygloat.constants.Message;
import com.polygloat.dtos.request.validators.ValidationError;
import com.polygloat.dtos.request.validators.ValidationErrorType;
import lombok.Getter;

import java.util.Collection;
import java.util.LinkedHashSet;
import java.util.Set;

public class ValidationException extends RuntimeException {
    @Getter
    private Set<ValidationError> validationErrors = new LinkedHashSet<>();

    public ValidationException(Message message, String... parameters) {
        this.validationErrors.add(new ValidationError(ValidationErrorType.CUSTOM_VALIDATION, message, parameters));
    }

    public ValidationException(ValidationErrorType type, String... parameters) {
        this.validationErrors.add(new ValidationError(type, Message.VALIDATION_ERROR, parameters));
    }

    public ValidationException(Collection<ValidationError> validationErrors) {
        this.validationErrors.addAll(validationErrors);
    }
}

File: backend/data/src/main/kotlin/io/tolgee/dtos/request/validators/exceptions/ValidationException.kt
package io.tolgee.dtos.request.validators.exceptions

import io.tolgee.constants.Message
import io.tolgee.dtos.request.validators.ValidationError
import io.tolgee.dtos.request.validators.ValidationErrorType

class ValidationException : RuntimeException {
  val validationErrors: MutableSet<ValidationError> = LinkedHashSet()

  constructor(message: Message, vararg parameters: String) {
    validationErrors.add(ValidationError(ValidationErrorType.CUSTOM_VALIDATION, message, *parameters))
  }

  constructor(validationErrors: Collection<ValidationError>?) {
    this.validationErrors.addAll(validationErrors!!)
  }
}
